src/main/java/org/skriptlang/skript/misc/colours/ColourUtils.java

```diff
@@ -0,0 +1,344 @@
+package org.skriptlang.skript.misc.colours;
+
+import ch.njol.skript.util.Color;
+import ch.njol.skript.util.ColorRGB;
+
+/**
+ * Utility class for colour manipulation and conversion.
+ */
+public class ColourUtils {
+
+	/**
+	 * Converts a {@link Color} to its hexadecimal representation.
+	 *
+	 * @param color the {@link Color} to convert
+	 * @return the hexadecimal string of the colour
+	 */
+	public static String toHex(Color color) {
+		return String.format("#%02X%02X%02X", color.getRed(), color.getGreen(), color.getBlue());
+	}
+
+	/**
+	 * Converts a {@link Color} to HSL (hue, saturation, lightness).
+	 *
+	 * @param color the {@link Color} to convert
+	 * @return a float array representing the HSL values
+	 */
+	public static float[] rgbToHsl(Color color) {
+		float r = color.getRed() / 255f;
+		float g = color.getGreen() / 255f;
+		float b = color.getBlue() / 255f;
+		float max = Math.max(r, Math.max(g, b));
+		float min = Math.min(r, Math.min(g, b));
+		float h, s, l = (max + min) / 2f;
+		if (max == min) {
+			h = s = 0f;
+		} else {
+			float delta = max - min;
+			s = l > 0.5f ? delta / (2f - max - min) : delta / (max + min);
+			if (max == r) {
+				h = ((g - b) / delta + (g < b ? 6f : 0f)) / 6f;
+			} else if (max == g) {
+				h = ((b - r) / delta + 2f) / 6f;
+			} else {
+				h = ((r - g) / delta + 4f) / 6f;
+			}
+		}
+		return new float[]{ h, s, l };
```

Comment:
This code and/or the other methods could use a few comments explaining what's happening
(and a link to the source if they were copied from somewhere)